CMakeLists.txt

```diff
@@ -108,7 +108,7 @@ if (PYBUILD)
   pybind11_add_module(_ext ${LAYOUT_SOURCES})
   set_target_properties(_ext PROPERTIES CXX_VISIBILITY_PRESET hidden)
   set_target_properties(_ext PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
-  target_link_libraries(_ext PRIVATE awkward-static)
+  target_link_libraries(_ext PRIVATE awkward)
```

Comment:
This is very likely the reason it's failing on Windows. Early in development, I encountered errors when linking any of the downstream libraries (libawkward.so or the Python extension module) to an upstream one that was not static. I think I also found references online that say the linking must be static in Windows. That's why I create static and dynamic libraries for each and do `target_link_libraries(X PRIVATE Y-static)` for each `X`, `Y`, and `Z` (i.e. all three platforms adhere to the least common denominator).

Is dynamic linking of the Python extension module necessary for your case? If not, could you set it back to linking against static? If so, could you put in an if-then-else to link against static only for Windows? (This feature of linking Python extensions is more important for Linux and MacOS than it is for Windows, so if this means that the feautre can be made to work for two out of three, that's acceptable.)

Reply:
@jpivarski I don't see a reason why this can't work in principle. That said, I can't contribute any windows solution at the moment. Using static linking for Windows should be fine.

I don't insist on using the dynamic linking. It should benefit the #211 which currently only affects hypothetical third-party python extensions linking to libawkward. Switching to dynamic linking will help to exercise the dynamic libawkward library, which is currently provided but is practically broken.